ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift

```diff
@@ -313,13 +313,22 @@ struct RoomTimelineItemFactory: RoomTimelineItemFactoryProtocol {
     
     private func aggregateReactions(_ reactions: [Reaction]) -> [AggregatedReaction] {
         reactions.map { reaction in
-            AggregatedReaction(accountOwnerID: userID, key: reaction.key, senders: reaction.senders.map(\.senderId))
+            let senders = reaction.senders
+                .map { senderData in
+                    ReactionSender(senderId: senderData.senderId, timestamp: Date(timeIntervalSince1970: TimeInterval(senderData.timestamp / 1000)))
+                }
+                .sorted { a, b in
+                    // Sort reactions most recent first
+                    a.timestamp > b.timestamp
+                }
+            return AggregatedReaction(accountOwnerID: userID, key: reaction.key, senders: senders)
         }
         .sorted { a, b in
-            // Sort by count and then by key for a consistence experience.
-            // Otherwise emojis can switch around. We can replace
-            // with timestamp as a secondary sort when it is available.
-            (a.count, a.key) > (b.count, b.key)
+            // Sort aggregated reactions by count and then by most recent reaction
+            if a.count == b.count {
+                return a.senders[0].timestamp < b.senders[0].timestamp
```

Comment:
Should this be the other way around for most recent?
```suggestion
                return a.senders[0].timestamp > b.senders[0].timestamp
```

Reply:
The comment should be clearer. It's sorted by count and then by aggregation with most recent reaction(index zero in the aggregation's list of reactions, which are in descending order from line 321) in ascending order(so new reactions are added to the end of the reaction layout).

I'll write that in a clearer way :)

Reply:
Right, understood 👍